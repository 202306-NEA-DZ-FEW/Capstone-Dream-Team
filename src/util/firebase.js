// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getFirestore } from "firebase/firestore";
import { getAuth } from "firebase/auth";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
    authDomain: "capstone-project-944c9.firebaseapp.com",
    projectId: "capstone-project-944c9",
    storageBucket: "capstone-project-944c9.appspot.com",
    messagingSenderId: "455661891383",
    appId: "1:455661891383:web:65bbc812478ffda076b4b2",
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const db = getFirestore(app);
export const auth = getAuth(app);
