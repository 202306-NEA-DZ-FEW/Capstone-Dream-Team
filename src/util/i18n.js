import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
