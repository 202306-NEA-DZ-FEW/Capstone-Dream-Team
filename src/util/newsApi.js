export async function fetchData(apiRoute) {
    const url = `https://newsapi.org/v2/${apiRoute}&apiKey=${process.env.NEXT_PUBLIC_NEWS_API}`;
    const response = await fetch(url);
    const data = await response.json();
    return data;
}
