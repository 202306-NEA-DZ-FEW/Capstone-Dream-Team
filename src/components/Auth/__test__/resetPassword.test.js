import renderer from "react-test-renderer";
import ResetPassword from "../resetPassword";
import { getAuth } from "@/util/firebase";

// Mock the Firebase services
jest.mock("@/util/firebase", () => ({
    getAuth: jest.fn(),
}));

jest.mock("next-i18next", () => ({
    useTranslation: jest.fn().mockReturnValue({ t: jest.fn() }),
}));

it("renders correctly", () => {
    getAuth.mockReturnValue();
    const tree = renderer.create(<ResetPassword />).toJSON();
    expect(tree).toMatchSnapshot();
});
