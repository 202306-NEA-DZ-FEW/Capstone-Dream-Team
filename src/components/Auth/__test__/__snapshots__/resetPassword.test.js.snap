// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col items-start justify-start pt-10 pr-10 pb-10 pl-10 bg-white shadow-2xl rounded-xl relative z-10"
>
  <h1
    className="w-full text-4xl  text-center leading-snug  mb-4"
  />
  <div
    className="w-full h-[49.21px] justify-center items-center inline-flex border rounded border-gray-300 ring-indigo-200 mb-4 shadow"
  >
    <div
      className="text-black text-m leading-tight "
    />
  </div>
  <form
    className="w-full mt-6 mr-0 mb-0 ml-0 relative space-y-8"
    onSubmit={[Function]}
  >
    <div
      className="relative"
    >
      <p
        className="bg-white pt-0 pr-2 pb-0 pl-2 -mt-3 mr-0 mb-0 ml-2 font-medium text-gray-600 absolute"
      />
      <svg
        className="absolute top-1/2 transform -translate-y-1/2 left-2 text-gray-400"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
        />
      </svg>
      <input
        className="border placeholder-gray-400 focus:outline-none focus:border-black w-full pt-4 pr-4 pb-4 pl-4 mt-2 mr-0 mb-0 ml-0 text-base block bg-white border-gray-300 rounded-md pl-8"
        onChange={[Function]}
        placeholder="123@ex.com"
        type="email"
        value=""
      />
    </div>
    <div>
      <button
        className="w-full inline-block pt-4 pr-5 pb-4 pl-5 text-xl font-medium text-center text-white bg-indigo-500 rounded-lg transition duration-200 hover:bg-indigo-600 ease"
        type="submit"
      />
    </div>
  </form>
  <div
    className="mb-6 flex text-m text-black mt-2"
  >
    <p>
       
    </p>
    <span
      className="ml-1 underline cursor-pointer"
      onClick={[Function]}
    >
       
    </span>
  </div>
</div>
`;
